sdks/unity/AgonesSdk.cs

```diff
@@ -0,0 +1,239 @@
+        /// The task result contains that the request is success or failure.
+        /// </returns>
+        public async Task<bool> SetAnnotation(string key, string value)
+        {
+            string json = JsonUtility.ToJson(new KeyValueMessage(key, value));
+            return await SendRequestAsync("/metadata/annotation", json, UnityWebRequest.kHttpVerbPUT);
+        }
+        #endregion
+
+        #region AgonesRestClient Private Methods
+        void Health()
+        {
+            _ = SendRequestAsync("/health", "{}");
+        }
+
+        async Task<bool> SendRequestAsync(string api, string json, string method = UnityWebRequest.kHttpVerbPOST)
+        {
+            // To prevent that an async method leaks after destroying this gameObject
+            cancellationTokenSource.Token.ThrowIfCancellationRequested();
+
+            var req = new UnityWebRequest(sidecarAddress + api, method)
+            {
+                uploadHandler = new UploadHandlerRaw(Encoding.UTF8.GetBytes(json)),
+                downloadHandler = new DownloadHandlerBuffer()
+            };
+            req.SetRequestHeader("Content-Type", "application/json");
+
+            await new AgonesAsyncOperationWrapper(req.SendWebRequest());
+
+            bool ok = req.responseCode == 200;
+
+            if (ok)
+            {
+                Log($"Agones SendRequest ok: {api} {req.downloadHandler.text}");
+            }
+            else
+            {
+                Log($"Agones SendRequest failed: {api} {req.error}");
+            }
+
+            return ok;
+        }
+
+        void Log(object message)
+        {
+            if (!logEnabled) { return; }
+
+#if UNITY_EDITOR
+            Debug.Log(message);
+#else
+            Console.WriteLine(message);
+#endif
+        }
+        #endregion
+
+        #region AgonesRestClient Nested Classes
+        class AgonesAsyncOperationWrapper
+        {
+            public UnityWebRequestAsyncOperation AsyncOp { get; }
+            public AgonesAsyncOperationWrapper(UnityWebRequestAsyncOperation unityOp)
+            {
+                AsyncOp = unityOp;
+            }
+
+            public AgonesAsyncOperationAwaiter GetAwaiter()
+            {
+                return new AgonesAsyncOperationAwaiter(this);
+            }
+        }
+
+        class AgonesAsyncOperationAwaiter : INotifyCompletion
+        {
+            private UnityWebRequestAsyncOperation asyncOp;
+            private Action continuation;
+
+            public AgonesAsyncOperationAwaiter(AgonesAsyncOperationWrapper wrapper)
+            {
+                asyncOp = wrapper.AsyncOp;
+                asyncOp.completed += OnRequestCompleted;
+            }
+
+            public bool IsCompleted => asyncOp.isDone;
+
+            public void GetResult()
+            {
+                asyncOp.completed -= OnRequestCompleted;
+
+                // remove references
+                asyncOp = null;

```

Comment:
Please document why this is necessary.

Reply:
Oh, It is unnecessary, I will remove it.